src/classes/CRLP_RollupProcessor.cls

```diff
@@ -547,6 +535,30 @@ public inherited sharing class CRLP_RollupProcessor {
         }
     }
 
+    private Boolean isSkewModeNeeded(SObject updatedRecord) {
+        if (isRollupFieldAtThreshold(updatedRecord)) {
+            updatedRecord.put(this.qualifiedSkewField, true);
+            return true;
+        }
+        return false;
+    }
+
+    private Boolean isRollupFieldAtThreshold(SObject summaryObject) {
+        Integer maxRelatedOppsForNonLDVMode = CRLP_Rollup_SVC.getMaxRelatedOppsForNonSkewMode();
+
+        if ((Decimal) summaryObject.get('npo02__NumberOfClosedOpps__c') >= maxRelatedOppsForNonLDVMode ||
+            (Decimal) summaryObject.get('npo02__NumberOfMembershipOpps__c') >= maxRelatedOppsForNonLDVMode) {
+            return true;
+
+        } else if (summaryObject.getSObjectType() == Contact.getSObjectType() &&
```

Comment:
This check for sobject type, which would be in a loop, falls into the category as why `useForceSkewModeField` is only instantiated once. The determination of whether this is for a Contact could be made earlier - once.  There's a couple of ways to do that too, so whatever feels best. 

I did do a quick test too. Calling `getSObjectType()` in a loop takes around 3x more milliseconds than just checking a boolean var in a. loop. Not a perfect test, but I wanted to make sure that the call to `getSobjectType()` was still relatively slow. Though, if my calculations are accurate, a rollup of 1M Contacts with a batch size of 200 would save around 80 seconds over the of the batch job in this one operation. I suppose combined with the previous checks for object for `useForceSkewMode` that would double to 160 seconds - nearly 3 minutes. Not a gigantic savings by any stretch. It's good data to have, and maybe 3 minutes over 1M Contacts is not worth the extra work to use a getter to limit the calls. So, I'll leave it up to the team to decide. FYI @dfuller114